docs/rfcs/001-recent-commits.md

```diff
@@ -0,0 +1,87 @@
+* Make it easy to undo most recent commit action, supersede amend check box.
+* Reinforce the visual "flow" of changes through being unstaged, staged, and now committed.
+* Provide a discoverable launch point for an eventual log feature to explore the full history.
+* Achieve greater consistency with GitHub desktop:
+
+![desktop](https://user-images.githubusercontent.com/7910250/36570484-1754fb3c-17e7-11e8-8da3-b658d404fd2c.png)
+
+## Explanation
+
+If the active repository has no commits yet, display a short panel with a background message: "Make your first commit".
+
+Otherwise, display a **recent commits** section containing a sequence of horizontal bars for each of the top three commits reachable from the current `HEAD`, with the most recently created commit on top. The user can resize the recent commits section. As it is expanded or shrunk, the number of visible commits is changed responsively.
+
+Each **recent commit** within the recent commits section summarizes that commit's metadata, to include:
+
+* GitHub avatar for both the committer and (if applicable) author. If either do not exist, show a placeholder.
+* The commit message (first line of the commit body) elided if it would be too wide.
+* A relative timestamp indicating how long ago the commit was created.
+
+On the most recent commit, display an "undo" button. Clicking "undo" performs a `git reset` and re-populates the commit message editor with the existing message.
+
+If any of the recent commits have been pushed to a remote, display a divider after the most recently pushed commit that shows an octocat icon. On hover, show the name of the remote tracking branch.
+
+Right-clicking a recent commit reveals a context menu offering interactions with the chosen commit. The context menu contains:
+
+* For the most recent commit only, an "Amend" option. "Amend" is enabled if changes have been staged or the commit message mini-editor contains text. Choosing this applies the staged changes and modified commit message to the most recent commit, in a direct analogue to using `git commit --amend` from the command line.
+* A "Revert" option. Choosing this performs a `git revert` on the chosen commit.
+* A "Hard reset" option. Choosing this performs a `git reset --hard` which moves `HEAD` and the working copy to the chosen commit. When chosen, display a modal explaining that this action will discard commits and unstaged working directory context. Extra security: If there are unstaged working directory contents, artificially perform a dangling commit, disabling GPG if configured, before enacting the reset. This will record the dangling commit in the reflog for `HEAD` but not the branch itself.
+* A "Soft reset" option. Choosing this performs a `git reset --soft` which moves `HEAD` to the chosen commit and populates the staged changes list with all of the cumulative changes from all commits between the chosen one and the previous `HEAD`.
+
+On click, reveal a tool-tip containing:
```

Comment:
In the future when we have a multi-file diff view, we can consider just opening a pane item with this information